### PR TITLE
Md button

### LINKED ISSUE #4: Material button
CSS for vaadin-button that makes it look like the default material design button: https://material.io/guidelines/components/buttons.html#buttons-style (search for "specs")

No additional style variations (dark theme, dense layout or raised style) or customization options (colors, fonts, sizes/spacing) are required at this point. All of those will be addressed by separate issues instead.

### PR DESCRIPTION
**Pull request by tomivirkki**
Closes #4

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-themes/12)

### CODE REVIEW

**Comment by tomivirkki** [comment]
Works for me. There are some known rough edges but since this is mostly a prototyping phase I think it's fine.

File: demo/material/button.html [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    </demo-navigation>
+
+    <h1>Button</h1>
+
+    <h3>Normal</h3>
+    <nice-demo-snippet>
+      <script type="text/html" slot="source">
+        <vaadin-button>Normal</vaadin-button>
+        <vaadin-button disabled>Disabled</vaadin-button>
+        <vaadin-button theme="primary">Primary</vaadin-button>
```
Comment (limonte) [inline demo/material/button.html]: `<vaadin-button theme="primary">`

We don't have the `primary` theme, this syntax is confusing. Why not `<vaadin-button primary>`?

**Comment by platosha** [comment]
:lgtm: overall. Have some non-blocking suggestions.

---

Reviewed 2 of 4 files at r1, 1 of 3 files at r4, 1 of 2 files at r6, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[material/colors.html, line 9 at r7](https://reviewable.io:443/reviews/vaadin/vaadin-themes/12#-Kku20Trqi8WDF-ypEo-:-Kku20Trqi8WDF-ypEo0:b-x9pkzl) ([raw file](https://github.com/vaadin/vaadin-themes/blob/09e994266f1da52521e67b1e2f9b3493d85b5200/material/colors.html#L9)):*
> ```HTML
> 
>       --material-primary-color: var(--primary-color, #2962FF);
>       --material-primary-contrast-color: var(--dark-theme-base-color, #fff);
> ```

BTW this name feels a bit off the other names, since `-contrast-color` is not use anywhere else, despite there are other color pairs.

Maybe use `-text-color` and `-background-color` suffixes for all the pairs? I mean:
- `--material-primary-color` stays as it is
- `--material-primary-background-color: var(--material-primary-color);` meaning backgrounds of `primary` zone
- `--material-primary-text-color: #fff;` meaning text color of `primary` zone

---

*[material/colors.html, line 11 at r7](https://reviewable.io:443/reviews/vaadin/vaadin-themes/12#-Kku0kqARt9mhJH41-KX:-Kku0kqARt9mhJH41-KY:b-58sqxl) ([raw file](https://github.com/vaadin/vaadin-themes/blob/09e994266f1da52521e67b1e2f9b3493d85b5200/material/colors.html#L11)):*
> ```HTML
>       --material-primary-contrast-color: var(--dark-theme-base-color, #fff);
> 
>       --material-disabled-color: #EAEAEA;
> ```

BTW Maybe `-disabled-background-color`, emphasising the difference from `-disabled-text-color`?

Would be even better to have the pair for `disabled` things as well. So that “disabled” thing gets `-disabled-background-color` and `-disabled-text-color`. 

---

*[material/elements/button.html, line 93 at r7](https://reviewable.io:443/reviews/vaadin/vaadin-themes/12#-Kku0BvcB6V9waUYKbcC:-Kku0BvcB6V9waUYKbcD:bqhhkz9) ([raw file](https://github.com/vaadin/vaadin-themes/blob/09e994266f1da52521e67b1e2f9b3493d85b5200/material/elements/button.html#L93)):*
> ```HTML
>       :host([raised]) {
>         background-color: var(--material-background-color);
>         box-shadow: 0 2px 2px rgba(0, 0, 0, 0.2), 0 0 0 1px rgba(0, 0, 0, 0.04);
> ```

BTW we should maybe relax the linter to allow multiple lines for `box-shadow` value.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/12#-:-Kku4_pFKfmj_e_mC87W:b-c1y6je)*

**Comment by limonte** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[material/elements/button.html, line 93 at r7](https://reviewable.io:443/reviews/vaadin/vaadin-themes/12#-Kku0BvcB6V9waUYKbcC:-Kku9FADWZeqbu9eiivZ:ba7iy1j) ([raw file](https://github.com/vaadin/vaadin-themes/blob/09e994266f1da52521e67b1e2f9b3493d85b5200/material/elements/button.html#L93)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

BTW we should maybe relax the linter to allow multiple lines for `box-shadow` value.
</blockquote></details>

The CSS linter is fine with this syntax, isn't it?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/12)*

**Comment by jouni** [comment]
> Maybe use -text-color and -background-color suffixes for all the pairs?

Seems like a nice idea, but it conflicts a bit with the feature of having separate properties for the primary color which is applied as a background and as a text color. For contrast reasons, sometimes those could be different.

**Comment by jouni** [comment]
I’m actually working on some improvements for this as we speak, so I’m just going to merge it and address these comments there.

**Pull request merged**